rpcs3/Emu/CPU/CPUTranslator.h

```diff
@@ -289,6 +289,24 @@ struct llvm_value_t<f64> : llvm_value_t<void>
 	}
 };
 
+#ifdef __APPLE__
```

Comment:
Not sure about this block, but I can't see why unsigned long needed specification since all the sized types are already handled above. Also why return 8-bit representation for 64-bit long? Or is it a compiler quirk?

Reply:
Not completely sure either. Apparently, clang treats unsigned long as a distinct type for the template specialization.

> Also why return 8-bit representation for 64-bit long? Or is it a compiler quirk?

oops, nice catch! :sweat_smile:

Reply:
This is not necessary, the issue is incompatibility between u64 and uptr on MacOS (unsigned long long vs unsigned long, same thing but compiler doesn't deal with it properly). Change uptr to uint64_t in types.h and Mac should build fine.